src/main/java/com/datatorrent/netlet/OptimizedEventLoop.java

```diff
@@ -59,7 +59,8 @@ public boolean add(SelectionKey key)
       if (key == null) {
         return false;
       }
-      if (pos >= keys.length) {
+      // An extra slot is needed to store the null delimiter when keys are retrieved
+      if (pos >= (keys.length - 1)) {
```

Comment:
Vlad has already implemented lambda function like loop. So probably if you
also look into it, it's gonna result in duplicate effort.

On Thursday, July 30, 2015, Pramod Immaneni <notifications@github.com>
wrote:

> In src/main/java/com/datatorrent/netlet/OptimizedEventLoop.java
> <https://github.com/DataTorrent/Netlet/pull/11#discussion_r35935121>:
>
> > @@ -59,7 +59,8 @@ public boolean add(SelectionKey key)
> >        if (key == null) {
> >          return false;
> >        }
> > -      if (pos >= keys.length) {
> > +      // An extra slot is needed to store the null delimiter when keys are retrieved
> > +      if (pos >= (keys.length - 1)) {
>
> Will look at both your suggestions and get back.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/DataTorrent/Netlet/pull/11/files#r35935121>.
>